### PR TITLE
Update rspec gems

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
<!--
## What was the end-user or developer problem that led to this PR?

Just making sure we use up to date gems. I was having some gem activation issues which I believe are somehow inherent to our complicated dev setup. I didn't dig too much and instead updated our dev gems to remove them.

## What is your fix for the problem, implemented in this PR?

Upgrade spec related gems.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [ ] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

**Review by simi — APPROVED** [review]

File: bundler/tool/bundler/dev_gems.rb.lock [hunk 1]
```diff
@@ -3,6 +3,7 @@ GEM
   specs:
     diff-lcs (1.5.0)
     hpricot (0.8.6)
+    hpricot (0.8.6-java)
```
Comment (simi) [inline bundler/tool/bundler/dev_gems.rb.lock:6]: Any idea why this was added? :thinking:

**Comment by simi** [comment]
It seems CI is unhappy with following error.

```
ArgumentError: wrong exec option symbol: exception
/Users/runner/work/rubygems/rubygems/Rakefile:12:in `system'
/Users/runner/work/rubygems/rubygems/Rakefile:12:in `bundle_dev_gemfile'
```

File: bundler/tool/bundler/dev_gems.rb.lock (hunk 1 above)
Comment (deivid-rodriguez) [reply bundler/tool/bundler/dev_gems.rb.lock:6]: No idea, but it seems correct, so maybe some bug fixed since this was last bundled?

**Comment by deivid-rodriguez** [comment]
Oh right, I used a feature of newer rubies only. I'll fix it.

**Pull request merged**